### DISCUSSION TITLE
GUI status 'Connection Failed'. Factory reset no working

### DISCUSSION THREAD

**Discussion by yoavli** [pyPhotometry/code#4] [Q&A]
GUI status says 'Connection Failed'. This board had previously worked on a different computer and a different pyphotometry board does work on this computer.
I tried tp do a factory reset of the filesystem, but the board LEDs do not light up when I press the USR and/or RST buttons.
When I connect the board with the USB connection both LEDs blink for few times and then do not respond anymore. The board is recognized by the computer, but the GUI status says 'Connection Failed'.

**Comment by ThomasAkam** [comment]
Hi Yoavli,

If factory resetting the board is not working and no LEDs light up when you press the usr or reset buttons on the pyboard, it is possible the pyboard has got damaged or failed.  Have you tried it on the computer where it was previously working?  If you look in the Device Manager (assuming your on Windows - I'm not sure what the Mac/Linux equivalent is), does the board show up under 'Ports (COM & LTP)?  It should show up either as a 'USB Serial Device' or Pyboard USB Comm port'.  If so does it report any error if you right click it and select properties?

**Reply by yoavli** [reply]
Thanks Thomas! Sorry for not providing all this info. the device also cannot connect to the pyPhotometry GUI when I connect it to the previous computer. The device is recognized in the device manager and everything in the 'properties' seems identical to another board that is working well. Are there any additional diagnostics we should do?

**Comment by ThomasAkam** [comment]
That is strange.   One way the board can get into a non-responsive state like that is if you try to update the version of micropython on the board, but accidentally upload a version that is for a different board (e.g. loading pyboard 1.0 firmware on a pyboard 1.1).  I assume you have not done this, but I wonder if the problem could still be something to do with the firmware on the board.  It would be worth trying to put the board into device firmware update (DFU) mode.  Normally you can do this by connecting the two pins highlighted in the image below and pressing the reset button.  The blue and red LEDs should go on and when you look in the device manager the board should show up as *STM device in DFU mode*.  If your able to get the board into DFU mode then you can try reloading the micropython firmware onto the board.  There are instruction for how to do this in the [Micropython windows setup guide](https://micropython.org/resources/Micro-Python-Windows-setup.pdf).

Let me know how you get on.

![image](https://user-images.githubusercontent.com/4661184/221691355-5fab645f-724d-40c9-920e-87a9e024fbc2.png)

**Reply by yoavli** [reply]
Thanks Thomas. Unfortunately connecting the two highlighted pins and pressing the reset button doesn't do anything to the board LEDs and to the status in the Device Manager. The only thing that makes the LEDs turn on and blink briefly is when I connect the board via USB to the computer. Other than that it's non-responsive...

**Comment by ThomasAkam** [comment]
It does sound like the pyboard has died, I'm not sure what might have caused this as they are usually very reliable.  The rest of the pyPhotometry board is likely fine so you should be able to get up and running by replacing the pyboard.  I'm not sure what OEPS's policy is on repairing boards that have stopped working but if you purchased the pyPhotometry board from them it would be worth asking.  Otherwise, replacing the pyboard is simple to do yourself as it is on a connector, but unfortunately they are out of stock at the micropython store due to the ongoing supply chain issues with the microcontroller.   There are knock-off pyboards available on sites like ebay but I have not tested any of these.

**Comment by yoavli** [comment]
Thanks Thomas! I will check all these options. The board should still be under the 1 year warranty of open-ephys.
